test/server.py

```diff
@@ -106,15 +107,23 @@ def _start_server(self, socket_handler):
         )
         self.server_thread.start()
         ready_event.wait()
+
         self.host = self.server_thread.host
         self.port = self.server_thread.port
         self.secure = self.server_thread.secure
 
     def get_connection(self):
         if self.h2:
-            return HTTP20Connection(self.host, self.port, self.secure)
+            if not self.proxy:
+                return HTTP20Connection(self.host, self.port, self.secure)
+            else:
+                return HTTP20Connection('http2bin.org', secure=self.secure, proxy_host=self.host, proxy_port=self.port)
```

Comment:
Sure. You are sticking to 80 width it appears right?  I trimmed a few other stragglers I saw that I had messed up as well.